packages/nx/src/generators/utils/project-configuration.ts

```diff
@@ -82,17 +80,73 @@ export function updateProjectConfiguration(
   projectName: string,
   projectConfiguration: ProjectConfiguration
 ): void {
+  if (
+    tree.exists(joinPathFragments(projectConfiguration.root, 'project.json'))
+  ) {
+    updateProjectConfigurationInProjectJson(
+      tree,
+      projectName,
+      projectConfiguration
+    );
+  } else if (
+    tree.exists(joinPathFragments(projectConfiguration.root, 'package.json'))
+  ) {
+    updateProjectConfigurationInPackageJson(
+      tree,
+      projectName,
+      projectConfiguration
+    );
+  } else {
+    throw new Error(
+      `Cannot update Project ${projectName} at ${projectConfiguration.root}. It either doesn't exist yet, or may not use project.json for configuration. Use \`addProjectConfiguration()\` instead if you want to create a new project.`
+    );
+  }
+}
+
+function updateProjectConfigurationInPackageJson(
+  tree: Tree,
+  projectName: string,
+  projectConfiguration: ProjectConfiguration
+) {
+  const packageJsonFile = joinPathFragments(
+    projectConfiguration.root,
+    'package.json'
+  );
+
+  const packageJson = readJson<PackageJson>(tree, packageJsonFile);
+
+  if (packageJson.name === projectConfiguration.name ?? projectName) {
```

Comment:
If the name is not equal, what do we do? Do we set it in `package.json#name` or do we set it in `package.json#nx.name`?

Reply:
nx.name, setting package.json name could break a published package